File: javascript/smartlist/smartlist-api-mysql-project/src/modules/auth/auth.response.codes.ts
import {
  BadRequestException,
  ForbiddenException,
  NotFoundException,
  UnauthorizedException,
} from '@nestjs/common';
import { AppResponseCodes } from 'src/app.response.codes';

export class AuthResponseCodes extends AppResponseCodes {
  public static USER_CREDENTIALS_INVALID: any = new BadRequestException(
    'user credentials are invalid',
  );

  public static USER_ACCOUNT_INACTIVE: any = new ForbiddenException(
    'user account is inactive',
  );

  public static USER_ACCOUNT_INVALID: any = new NotFoundException(
    'invalid user account',
  );

  public static ACCESS_TOKEN_INVALID: any = new UnauthorizedException(
    'invalid access token',
  );
}

File: java/smartlist/src/main/java/com/smartlist/module/auth/AuthResponseCodes.java
package com.smartlist.module.auth;

import com.smartlist.utils.AppResponseCodes;
import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

public class AuthResponseCodes extends AppResponseCodes {

    private AuthResponseCodes(){}

    public static final String USER_CREDENTIALS_INVALID_DESCRIPTION = "user credentials are invalid";
    public static final String USER_ACCOUNT_INACTIVE_DESCRIPTION = "user account is inactive";
    public static final String USER_ACCOUNT_INVALID_DESCRIPTION = "invalid user account";
    public static final String ACCESS_TOKEN_INVALID_DESCRIPTION = "invalid access token";


    public static ResponseStatusException USER_CREDENTIALS_INVALID = new ResponseStatusException(HttpStatus.BAD_REQUEST, USER_CREDENTIALS_INVALID_DESCRIPTION);

    public static ResponseStatusException USER_ACCOUNT_INACTIVE = new ResponseStatusException(HttpStatus.FORBIDDEN, USER_ACCOUNT_INACTIVE_DESCRIPTION);

    public static ResponseStatusException USER_ACCOUNT_INVALID = new ResponseStatusException(HttpStatus.NOT_FOUND, USER_ACCOUNT_INVALID_DESCRIPTION);

    public static ResponseStatusException ACCESS_TOKEN_INVALID = new ResponseStatusException(HttpStatus.UNAUTHORIZED, ACCESS_TOKEN_INVALID_DESCRIPTION);



}
